### PR TITLE
feat: Add dockerfile

### PR DESCRIPTION
**Pull request by mikeldking**
This adds the ability to run the server in a docker container.

```shell
docker build -t phoenix .
docker run -d --name phoenix -p 6060:6060 phoenix
```

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+docs/
+examples/
+dist/
+integration-tests/
+scripts/
+.github
+.tours
```
Comment (axiomofjoy) [inline .dockerignore:7]: This is missing `tests`. It might be easier to copy `src` rather than ignore everything else.

**Review by axiomofjoy — APPROVED** [review]
Really cool.

**Comment by axiomofjoy** [comment]
Is the plan to follow up with volume mounts and a way to ensure we don't just indefinitely accumulate traces in memory?

**Comment by mikeldking** [comment]
> Is the plan to follow up with volume mounts and a way to ensure we don't just indefinitely accumulate traces in memory?

Yeah we have to scope that out.

File: .dockerignore (hunk 1 above)
Comment (mikeldking) [reply .dockerignore:7]: we could but having an ignore is valuble too

**Pull request merged**